File: src/Examples/CheckbotBimLink/BimApiLinkFeaExample/Importers/CrossSectionImporter.cs
using BimApiLinkFeaExample.BimApi;
using IdeaStatiCa.BimApiLink.Importers;
using IdeaStatiCa.BimApi;

namespace BimApiLinkFeaExample.Importers
{
	internal class CrossSectionImporter : IntIdentifierImporter<IIdeaCrossSection>
	{
		public CrossSectionImporter(/*TODO pass API using DI*/)
		{
		}

		public override IIdeaCrossSection Create(int id)
		{
			return new CrossSectionByName(id)
			{
				MaterialNo = 1,
				Name = "IPE200",
			};
		}
	}
}

File: examples/bimapi/CheckbotBimLink/CppFeaExample/CppFeaApiWrapper/CrossSectionImporter.cpp
#include "pch.h"
#include "CrossSectionImporter.h"
#include "CrossSectionByName.h"


namespace CppFeaApiWrapper
{
	namespace Importers
	{
		CrossSectionImporter::CrossSectionImporter(ImporterContext^ context)
		{
			this->context = context;
		}

		IIdeaCrossSection^ CrossSectionImporter::Create(int id)
		{
			CppFeaApiWrapper::BimApi::CrossSectionByName^ css = gcnew CppFeaApiWrapper::BimApi::CrossSectionByName(id);
			css->MaterialNo = 1;
			css->Name = "IPE200";

			return css;
		}
	}
}